src/producer.cc

```diff
@@ -339,6 +348,78 @@ void Producer::ConfigureCallback(const std::string &string_key, const v8::Local<
   }
 }
 
+Baton Producer::InitTransactions(int32_t timeout_ms) {
+  if (!IsConnected()) {
+    return Baton(RdKafka::ERR__STATE);
+  }
+
+  RdKafka::Producer* producer = dynamic_cast<RdKafka::Producer*>(m_client);
+  RdKafka::Error* error = producer->init_transactions(timeout_ms);
+
+  if ( NULL == error) {
+    return Baton(RdKafka::ERR_NO_ERROR);
+  }
+  else {
+    Baton result(error->code());
+    delete error;
+    return result;
+  }
+}
+
+Baton Producer::BeginTransaction() {
+  if (!IsConnected()) {
+    return Baton(RdKafka::ERR__STATE);
+  }
+
+  RdKafka::Producer* producer = dynamic_cast<RdKafka::Producer*>(m_client);
+  RdKafka::Error* error = producer->begin_transaction();
+
+  if ( NULL == error) {
+    return Baton(RdKafka::ERR_NO_ERROR);
+  }
+  else {
+    Baton result(error->code());
+    delete error;
+    return result;
+  }
+}
+
+Baton Producer::CommitTransaction(int32_t timeout_ms) {
+  if (!IsConnected()) {
+    return Baton(RdKafka::ERR__STATE);
+  }
+
+  RdKafka::Producer* producer = dynamic_cast<RdKafka::Producer*>(m_client);
+  RdKafka::Error* error = producer->commit_transaction(timeout_ms);
+
+  if ( NULL == error) {
+    return Baton(RdKafka::ERR_NO_ERROR);
+  }
+  else {
+    Baton result(error->code());
```

Comment:
The librdkafka transactional API uses the new richer `rd_kafka_error_t` type that provides some extra error attributes.
These attributes, fatal, txn_requires_abort and retriable, are necessary to expose in this node.js API as well since it tells the application how the errors are to be handled, which can not be derrived from the error code itself.

Reply:
Thank you for your feedback, I have addressed it. I am now exposing a richer JS object populated with data from `rd_kafka_error_t`.